src/main/java/org/commcare/xml/ResourceParser.java

```diff
@@ -46,6 +47,6 @@ public Resource parse() throws InvalidStructureException, IOException, XmlPullPa
             }
         }
 
-        return new Resource(version, id, locations, descriptor);
+        return new Resource(version, id, locations, descriptor, lazy != null && lazy.contentEquals("true"));
```

Comment:
`"true".contentEquals(lazy)` can be used to avoid null checks.

Reply:
"true".contentEquals(null) does throw a NPE

Reply:
Ohh, sorry for that! Just curious why are  we using `contentEquals` over `equals`?

Reply:
no particular reason, they both are same for this use case. `contentEquals` is a more free implementation of `equals` allowing to compare across different `CharSequence` implementations so I just default to it for any string comparison.